ui.frontend/src/view/FormFieldBase.js

```diff
@@ -463,24 +463,22 @@ class FormFieldBase extends FormField {
      * Updates the HTML class based on the existence of a value in a field.
      */
     updateEmptyStatus() {
-        if (!this.getWidget())
-            return;
-
-        const updateModifierClass = (widget, newValue) => {
-            const widgetBemClass = widget.className.split(/\s+/).filter(bemClass => bemClass.endsWith("__widget"))[0];
-            const filledModifierClass = `${widgetBemClass}--filled`;
-            const emptyModifierClass = `${widgetBemClass}--empty`;
-
-            widget.classList.add(newValue ? filledModifierClass : emptyModifierClass);
-            widget.classList.remove(newValue ? emptyModifierClass : filledModifierClass);
-        };
-
+        if (!this.getWidget()){
+          return;
+        }
+        let value = '';
+        const checkedWidget = ['radio', 'checkbox'];
         // radiobutton, checkbox, datefield(AFv1, not datepicker), etc. have multiple widgets in the form of a NodeList
         if (this.widget instanceof NodeList) {
-            this.widget.forEach((widget) => updateModifierClass(widget, (widget.type === "radio" || widget.type === "checkbox") ? widget.checked : widget.value))
+          value = Array.from(this.widget).map((widget) => checkedWidget.includes(widget.type) ? widget.checked : widget.value).find(value => value);
         } else {
-            updateModifierClass(this.widget, this.widget.value)
+          value = checkedWidget.includes(this.widget.type) ? this.widget.checked : this.widget.value;
         }
+        const bemClass = Array.from(this.element.classList).filter(bemClass => !bemClass.includes('--'))[0]
```

Comment:
Can you add a test case for the existing of this classe based on empty /filled value in runtime ?

Reply:
done